test/weights.jl

```diff
@@ -261,7 +261,48 @@ end
     @test_throws ErrorException median(data, f(wt))
 end
 
-@testset "Quantile $f" for f in weight_funcs
+
+# Quantile fweights
+@testset "Quantile fweights" begin
+    data = (
+        [7, 1, 2, 4, 10],
+        [7, 1, 2, 4, 10],
+        [7, 1, 2, 4, 10, 15],
+        [1, 2, 4, 7, 10, 15],
+        [0, 10, 20, 30],
+        [1, 2, 3, 4, 5],
+        [1, 2, 3, 4, 5],
+        [30, 40, 50, 60, 35],
+        [2, 0.6, 1.3, 0.3, 0.3, 1.7, 0.7, 1.7],
+        [1, 2, 2],
+        [3.7, 3.3, 3.5, 2.8],
+        [100, 125, 123, 60, 45, 56, 66],
+        [2, 2, 2, 2, 2, 2],
+        [2.3],
+        [-2, -3, 1, 2, -10],
+        [1, 2, 3, 4, 5],
+        [5, 4, 3, 2, 1],
+        [-2, 2, -1, 3, 6],
+        [-10, 1, 1, -10, -10],
+    )
+    p = [0.0, 0.25, 0.5, 0.75, 1.0]
+    for x in data
+        @test quantile(x, fweights(ones(Int64, length(x))), p) ≈ quantile(x, p)
```

Comment:
It would be nice to test more thoroughly the results by combining various inputs with various weights, as done below with non frequency weights. You could use a `rep` helper function (https://github.com/JuliaLang/julia/issues/16443) to generate vectors with repeated entries to call the unweighted `quantile` method on.

Also, there aren't any zeros no negative non-integer values. Wouldn't hurt to add some.